integration-tests/testutils/report/parser.go

```diff
@@ -40,43 +42,52 @@ var (
 	skipRegexp     = regexp.MustCompile(fmt.Sprintf(commonPattern, "SKIP") + skipPatternSufix)
 )
 
-// ParserReporter is a type implementing io.Writer that
-// parses the input data and sends the results to the Next
-// reporter
+// Statuser reports the status of a test.
+type Statuser interface {
+	Status(subunit.Event) error
+}
+
+// SubunitV2ParserReporter is a type that parses the input data
+// and sends the results as a test status.
 //
 // The input data is expected to be of the form of the textual
 // output of gocheck with verbose mode enabled, and the output
-// will be of the form defined by the subunit format before
-// the binary encoding. There are constants reflecting the
-// expected patterns for this texts.
+// will be of the form defined by the subunit v2 format There
+// are constants reflecting the expected patterns for this texts.
 // Additionally, it doesn't take  into account the SKIPs done
 // from a SetUpTest method, due to the nature of the snappy test
 // suite we are using those for resuming execution after a reboot
 // and they shouldn't be reflected as skipped tests in the final
 // output. For the same reason we use a special marker for the
 // test's announce.
-type ParserReporter struct {
-	Next io.Writer
+type SubunitV2ParserReporter struct {
+	statuser Statuser
```

Comment:
It's ok to remove it, at first I figured out the process as a pipeline, each stage with the same structure, do the work in the write method and pass the ball to the next step in the chain (represented by the Next field), with all the steps implementing the io.Writer interface. This made sense with three steps: parsing -> binary encoding -> file writing, and fitted well with the io.MultiWriter at base_test that guarantees that we keep getting our regular output on stdout.

With just two steps (subunit stream generation, which involves parsing -> file writing) maybe it's better to pass the file writer as an argument, as you wish. We could also keep separated the parsing stage from the encoding stage by wrapping the Status instance in something that implements the writer interface, or even get rid of all this and do it a little better using channels and a structure of connected generators instead of sending the whole strings to write.